src/utils/vc/crypto/Bls12381BBSSignatureDock2022.js

```diff
@@ -0,0 +1,193 @@
+} from '@docknetwork/crypto-wasm-ts/lib/anonymous-credentials';
+
+import {
+  initializeWasm,
+  SignatureG1,
+  BBSPlusSecretKey,
+  getSigParamsOfRequiredSize,
+} from '@docknetwork/crypto-wasm-ts';
+
+import jsonld from 'jsonld';
+import { SECURITY_CONTEXT_URL } from 'jsonld-signatures';
+import { Bls12381BBSSigDockSigName } from './constants';
+
+import Bls12381BBSVerificationKeyDock2022 from './Bls12381BBSVerificationKeyDock2022';
+import CustomLinkedDataSignature from './custom-linkeddatasignature';
+
+const SUITE_CONTEXT_URL = 'https://www.w3.org/2018/credentials/v1';
+
+const DEFAULT_PARSING_OPTS = {
+  useDefaults: true,
+};
+
+/**
+ * A BBS+ signature suite for use with BLS12-381 Dock key pairs
+ */
+export default class Bls12381BBSSignatureDock2022 extends CustomLinkedDataSignature {
+  /**
+   * Default constructor
+   * @param options {SignatureSuiteOptions} options for constructing the signature suite
+   */
+  constructor(options = {}) {
+    const {
+      verificationMethod, signer, keypair, verifier,
+    } = options;
+
+    super({
+      type: Bls12381BBSSigDockSigName,
+      LDKeyClass: Bls12381BBSVerificationKeyDock2022,
+      contextUrl: SUITE_CONTEXT_URL,
+      alg: 'ES256K',
+      signer: signer || Bls12381BBSSignatureDock2022.signerFactory(keypair, verificationMethod),
+      verifier,
+    });
+
+    this.proof = {
+      '@context': [
+        {
+          sec: 'https://w3id.org/security#',
+          proof: {
+            '@id': 'sec:proof',
+            '@type': '@id',
+            '@container': '@graph',
+          },
+        },
+        'https://ld.dock.io/security/bbs/v1',
+      ],
+      type: Bls12381BBSSigDockSigName,
+    };
+
+    this.verificationMethod = verificationMethod;
+    if (keypair) {
+      if (verificationMethod === undefined) {
+        this.verificationMethod = keypair.id;
+      }
+      this.key = keypair;
+    }
+  }
+
+  /**
+   * @param {object} options - The options to use.
+   * @param {object} options.document - The document to be signed/verified.
+   * @param {object} options.proof - The proof to be verified.
+   * @param {function} options.documentLoader - The document loader to use.
+   * @param {function} options.expansionMap - NOT SUPPORTED; do not use.
+   *
+   * @returns {Promise<{Uint8Array}>}.
+   */
+  async createVerifyData(options) {
+    await initializeWasm();
+
+    // Serialize the data for signing
+    const serializedCredential = await this.serializeForSigning(options);
+
+    // Create an encoder through the schema, if one exists
+    let credSchema;
+    const { credentialSchema } = options.document;
+    if (credentialSchema) {
+      credSchema = CredentialSchema.fromJSON({
+        ...credentialSchema,
+        parsingOptions: {
+          ...DEFAULT_PARSING_OPTS,
+          ...(credentialSchema.parsingOptions || {}),
+        },
+      });
+    } else {
+      throw new Error('Credential must have define credentialSchema');
```

Comment:
"must have define" -> "must define"

Reply:
was meant to be `must have defined` but ill go with yours